controllers/osbuildconfig_controller_test.go

```diff
@@ -0,0 +1,575 @@
+				// given
+				osbuildConfigInstance.Spec.Details.TargetImage.TargetImageType = osbuildv1alpha1.EdgeContainerImageType
+				osBuildConfigRepository.EXPECT().PatchStatus(requestContext, osbuildConfigInstance, gomock.Any()).Return(nil)
+
+				// when
+				result, err := reconciler.Reconcile(requestContext, request)
+
+				// then
+				Expect(err).To(BeNil())
+				Expect(result).To(Equal(resultDone))
+			})
+
+			It("should Done if last osbuild instance build type is edge installer", func() {
+				// given
+				osbuildConfigInstance.Spec.Details.TargetImage.TargetImageType = osbuildv1alpha1.EdgeInstallerImageType
+				edgeInstallerTargetType := osbuildv1alpha1.TargetImageType("edge-installer")
+				osbuildConfigInstance.Status.LastBuildType = &edgeInstallerTargetType
+				osBuildConfigRepository.EXPECT().PatchStatus(requestContext, osbuildConfigInstance, gomock.Any()).Return(nil)
+
+				// when
+				result, err := reconciler.Reconcile(requestContext, request)
+
+				// then
+				Expect(err).To(BeNil())
+				Expect(result).To(Equal(resultDone))
+			})
+
+			Context("should create new OSBuild instance if target image type is edge installer and last build type is edge-container", func() {
+				BeforeEach(func() {
+					osbuildConfigInstance.Spec.Details.TargetImage.TargetImageType = osbuildv1alpha1.EdgeInstallerImageType
+					edgeContainerTargetType := osbuildv1alpha1.TargetImageType("edge-container")
+					osbuildConfigInstance.Status.LastBuildType = &edgeContainerTargetType
+				})
+
+				It("and fail on patch status", func() {
+					// given
+					osBuildConfigRepository.EXPECT().PatchStatus(requestContext, osbuildConfigInstance, gomock.Any()).Return(errFailed)
+
+					// when
+					result, err := reconciler.Reconcile(requestContext, request)
+
+					// then
+					Expect(err).To(BeNil())
+					Expect(result).To(Equal(resultShortRequeue))
+				})
+
+				It("and fail on create OSBuild instance", func() {
+					// given
+					osBuildConfigRepository.EXPECT().PatchStatus(requestContext, osbuildConfigInstance, gomock.Any()).Return(nil).Times(2)
+					osBuildCRCreator.EXPECT().Create(requestContext, osbuildConfigInstance, osbuildv1alpha1.EdgeInstallerImageType).Return(errFailed)
+
+					// when
+					result, err := reconciler.Reconcile(requestContext, request)
+
+					// then
+					Expect(err).To(BeNil())
+					Expect(result).To(Equal(resultShortRequeue))
+				})
+
+				It("and requeue", func() {
+					// given
+					osBuildConfigRepository.EXPECT().PatchStatus(requestContext, osbuildConfigInstance, gomock.Any()).Return(nil).Times(2)
+					osBuildCRCreator.EXPECT().Create(requestContext, osbuildConfigInstance, osbuildv1alpha1.EdgeInstallerImageType).Return(nil)
+
+					// when
+					result, err := reconciler.Reconcile(requestContext, request)
+
+					// then
+					Expect(err).To(BeNil())
+					Expect(result).To(Equal(resultLongRequeue))
+				})
+
+			})
+		})
+
+	})
+
+	Context("OSBuildConfig status is not Progressing state", func() {
```

Comment:
Sorry, but, I don't understand what's going on in this `Context`, not why these are the expected behaviors (e.g. a new instance is created when in Failed status, i.e. retry) or where it is implemented

Reply:
It's an old test - as I wrote -I need to fix the UT, I will push it today